File: NewsDaily2/app/src/main/java/com/andruid/magic/newsdaily/util/MediaUtil.java
package com.andruid.magic.newsdaily.util;

import android.net.Uri;
import android.os.Bundle;
import android.support.v4.media.MediaDescriptionCompat;
import android.support.v4.media.MediaMetadataCompat;

import com.andruid.magic.newsdaily.model.AudioNews;

public class MediaUtil {
    public static MediaDescriptionCompat getMediaDescription(AudioNews audioNews){
        Bundle extras = new Bundle();
        extras.putParcelable(MediaMetadataCompat.METADATA_KEY_ALBUM_ART_URI,
                Uri.parse(audioNews.getNews().getImageUrl()));
        extras.putParcelable(MediaMetadataCompat.METADATA_KEY_DISPLAY_ICON_URI,
                Uri.parse(audioNews.getNews().getImageUrl()));
        return new MediaDescriptionCompat.Builder()
                .setMediaId(audioNews.getUri())
                .setIconUri(Uri.parse(audioNews.getNews().getImageUrl()))
                .setTitle(audioNews.getNews().getTitle())
                .setDescription(audioNews.getNews().getDesc())
                .setExtras(extras)
                .build();
    }

    public static MediaMetadataCompat buildMetaData(AudioNews audioNews) {
        return new MediaMetadataCompat.Builder()
                .putString(MediaMetadataCompat.METADATA_KEY_ALBUM_ART_URI,
                        audioNews.getNews().getImageUrl())
                .putString(MediaMetadataCompat.METADATA_KEY_ALBUM, audioNews.getNews().getSourceName())
                .putString(MediaMetadataCompat.METADATA_KEY_TITLE, audioNews.getNews().getTitle())
                .build();
    }
}

File: NewsDaily/app/src/main/java/com/death14stroke/newsdaily/ui/util/MediaUtil.kt
package com.death14stroke.newsdaily.ui.util

import android.net.Uri
import android.support.v4.media.MediaDescriptionCompat
import android.support.v4.media.MediaMetadataCompat
import androidx.core.os.bundleOf
import com.death14stroke.newsdaily.data.model.AudioNews
import com.google.android.exoplayer2.MediaItem

fun AudioNews.getMediaDescription(): MediaDescriptionCompat {
    val extras = bundleOf(
        MediaMetadataCompat.METADATA_KEY_ALBUM_ART_URI to news.imageUrl,
        MediaMetadataCompat.METADATA_KEY_DISPLAY_ICON_URI to news.imageUrl,
        MediaMetadataCompat.METADATA_KEY_ALBUM to news.source.name
    )
    return MediaDescriptionCompat.Builder()
        .setMediaId(uri)
        .setIconUri(Uri.parse(news.imageUrl ?: ""))
        .setTitle(news.title)
        .setDescription(news.desc)
        .setExtras(extras)
        .build()
}

private fun MediaItem.toAudioNews() = localConfiguration?.tag as AudioNews?

fun MediaItem.getMediaDescription(): MediaDescriptionCompat =
    toAudioNews()?.getMediaDescription() ?: MediaDescriptionCompat.Builder().build()

fun MediaItem.getTitle() = toAudioNews()?.news?.title

fun MediaItem.getSource() = toAudioNews()?.news?.source?.name

fun MediaItem.getAlbumArtUri() = toAudioNews()?.news?.imageUrl